cmd/thanos/query.go

```diff
@@ -304,7 +304,9 @@ func runQuery(
 					}
 					fileSDCache.Update(update)
 					stores.Update(ctxUpdate)
-					dnsProvider.Resolve(ctxUpdate, append(fileSDCache.Addresses(), storeAddrs...))
+					if err := dnsProvider.Resolve(ctxUpdate, append(fileSDCache.Addresses(), storeAddrs...)); err.Err() != nil {
```

Comment:
There is quite big bug here, but CI should help you to find what. 

I will not say exactly what's wrong for educational purpose, please try to get from CI what's wrong you should be able to quickly spot this! :hugs: Let me know if you are blocked.

Reply:
Challenge accepted!